snuba/reader.py

```diff
@@ -25,7 +25,12 @@
     Row = MutableMapping[str, Any]
     Result = TypedDict(
         "Result",
-        {"meta": Sequence[Column], "data": Sequence[Row], "totals": Row},
+        {
+            "meta": Sequence[Column],
+            "data": Sequence[Row],
+            "totals": Row,
+            "applied_sampling_rate": Optional[float],
```

Comment:
Returning only the rate and no information around how that was calculated is an intentional decision. The production code does not need anything else to decide how to treat the result. Any additional information should not impact how the client processes the results, having the bare minimum the client need available in prod makes it harder to use the results improperly.
Specifically the client should only take one decision:
- if they provided fixed, none, or adaptive without adjustment mode, the client should set expectations to the user if sampling is not 1.0 and apply the proper aggregation adjustment
- IF they provided adaptive with adjustment mode, then the client knows they must not touch the results and just set accuracy expectations to the user if sampling is not 1.0. 
Any additional field could be useful in debug mode but never used in prod.

Reply:
Still thinking about this. I don't think I agree.

> * IF they provided adaptive with adjustment mode, then the client knows they must not touch the results and just set accuracy expectations to the user if sampling is not 1.0.

I think this puts an undue burden on the caller.